### PR TITLE
chore: java engine refactor

### PR DESCRIPTION
**Pull request by gastonfournier**
- Remove indirection level (lib)
- Adapt to new method names
- Tests don't pass the specs (`[01-simple-examples.json] Failed test 'Feature.B should be disabled': expected false, got true ==> expected: <false> but was: <true>`) but I'll look into this in another PR

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -8,7 +8,7 @@ Yggdrasil is a Rust project designed to create the core of the Unleash SDK domai
 
 ## Building the Core
 
-Easy enough - run `cargo build` from the root of the project. You'll need an up to date set of Rust tools to do this.
+Easy enough - run `cargo build --release` from the root of the project. You'll need an up to date set of Rust tools to do this.
```
Comment (sighphyre) [inline README.md:11]: :heart:

File: java-engine/src/main/java/io/getunleash/engine/UnleashEngine.java [hunk 2]
```diff
@@ -18,45 +19,46 @@ interface UnleashFFI extends Library {
...
 }
 
 public class UnleashEngine {
     private Pointer ptr;
     private ObjectMapper mapper = new ObjectMapper();
 
     public UnleashEngine() {
-        ptr = UnleashFFI.INSTANCE.engine_new();
+        ptr = UnleashFFI.INSTANCE.new_engine();
+        mapper.configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
```
Comment (gastonfournier) [inline java-engine/src/main/java/io/getunleash/engine/UnleashEngine.java:41]: This is just for being resilient when adding new fields that come from the engine state

Comment (sighphyre) [reply java-engine/src/main/java/io/getunleash/engine/UnleashEngine.java:41]: Nice!

**Review by sighphyre — APPROVED** [review]
G

**Pull request merged**